[CS2103T-F13-1] LiBerry

shiyao821 opened the pull request:
@linyutinglyt
@hoholyin
@Cronyxx
@seeziyang

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
-[width="59%",cols="22%,<23%,<25%,<30%",options="header",]
+[width="75%",cols="22%,<23%,<25%,<30%",options="header",]
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
```
The format should be "As a ... I can .... So that I can..."

hoholyin replied:
Actually, this format is also acceptable. In fact, it gives us more flexibility to express different user stories, allowing us to cover more user stories.
https://www.atlassian.com/agile/project-management/user-stories

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
-|`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
+|`* * *` |librarian |add a book brought/donated by people to the library |keep track of the addition of the book
```
The benefit sounds like a repeat of the function. Can be rephrased as "I can maintain a record of all the books in the library"

hoholyin replied:
Adding a book and keeping track of the book's status is different.

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
+|`* * *` |librarian |add a book brought/donated by people to the library |keep track of the addition of the book
 
-|`* * *` |user |add a new person |
+|`* * *` |librarian |delete books that are no longer available |keep track of the deletion of the book
```
Same as previous comment

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
+|`* * *` |forgetful librarian |mark a book as loaned out |tell borrowers if the book is available
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * *` |forgetful librarian |mark a book as returned |let borrowers know that the book will now be available
```
The status of the book should just be "loaned" or "not loaned" to make the status of the book more clear

hoholyin replied:
As a user story, it has to sound natural.

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
+
+|`* * *` |meticulous librarian |record the movement of books in and out |keep track of available books here
+
+|`* * *` |helpful librarian |assist a borrower to registering in the system |help new borrowers start borrowing books
```
It is unclear if the librarian is helping the borrower with the process of registering or if the librarian is registering the borrower himself

hoholyin replied:
User story is not the same as implementation. More details will be provided in the implementation/use case.

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
+
+|`* *` |careless librarian |be able to undo a command |undo my input mistakes
+
+|`* *` |careless librarian |be able to redo a command |undo my undo commands, in case I need it, without having to
```
Repeated user story

hoholyin replied:
One is redo command, one is undo command

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
+|`* *` |health conscious, night-working librarian |change the user interface into a night mode |reduce the impact
+of light and glare on my eyes when I am working at night
+
+|`* *` |impatient librarian |have my command inputs returned within 1 sec |serve my customers quickly
```
This should be non-functional requirement

hoholyin replied:
A non-functional requirement can be in the form of a user story in case you are unaware.
https://nus-cs2103-ay1920s1.github.io/website/se-book-adapted/chapters/specifyingRequirements.html#usage

limhawjia commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +279,107 @@ Refer to the guide <<DevOps#, here>>.
+|`* *` |helpful librarian |be able to reserve a currently on-loan book |allow borrowers to borrow the book once it
+is returned
+
+|`* *` |librarian |be able to see an image of the book cover |be reminded of the commands available
```
Function does not match benefit

hoholyin replied:
Thank you for pointing this out, we will amend it.

limhawjia reviewed (commented):
Good job with the use cases and NFRs, but some of the user stories can be phrased better.

KendrickAng commented on docs/DeveloperGuide.adoc:
```diff
@@ -218,29 +328,72 @@ image::CommitActivityDiagram.png[]
+Given below is the sequence diagram of the generation of loan slip during the loan of a book.
+
+.Sequence Diagram for the generation of a loan slip
+image::LoanSlipGeneration.png[]
```
The return arrow on a sequence diagram should be dotted. There's a way to place dotted lines using PlantUml!

hoholyin replied:
Good catch! Will make ammendments to it

KendrickAng commented on docs/DeveloperGuide.adoc:
```diff
@@ -88,61 +106,78 @@ The `UI` component,
 === Model component
 
 .Structure of the Model Component
 image::ModelClassDiagram.png[]
```
Perhaps the multiplicity for Tag could be included too, for consistency.

hoholyin replied:
The multiplicity is included in the diagram, but it may not be obvious since it is blocked by the `Tag` title. Will try to use hidden arrows to expose it.

KendrickAng reviewed (commented):
Good job! However, we found some inconsistencies for the UML diagrams. Please fix them!

shiyao821 committed:
fix: ResultDisplay UI and scrolling

seeziyang committed:
Merge pull request #261 from Cronyxx/bugfix-#243

Bugfix #243

seeziyang committed:
Merge pull request #260 from hoholyin/info-command

Delete command bug fixes and tests

shiyao821 committed:
fix: UI Resizing Issues

seeziyang committed:
Add reponsense link

shiyao821 committed:
fix: Info Window toggleui

shiyao821 committed:
fix: checkstyles

shiyao821 committed:
fix: synced Help Window UI theme

hoholyin committed:
Merge branch 'master' into master

linyutinglyt committed:
edit borrwer ID part

linyutinglyt committed:
edit borrower part

hoholyin committed:
Merge pull request #266 from shiyao821/master

UI Dark Theme, Resizing, and updates

hoholyin committed:
Merge pull request #262 from linyutinglyt/ugdg

Update UG and DG

hoholyin committed:
Fix bug (Unregister serving borrower)

hoholyin committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main into portfolio

seeziyang committed:
Merge branch 'master' into update-ppp

seeziyang committed:
Update javafx version

Cronyxx committed:
refactor and add tests

-Changed CommandHistory to CommandHistoryManager.
-Created CommandHistory interface.
-Changed undo and redo methods in CommandHistoryManager to getUndoCommand and getRedoCommand.
-Added assertions and logging to CommandHistoryManager.
-Created tests for CommandHistoryManager.

Cronyxx committed:
update tests in ModelManagerTest

Cronyxx committed:
update AddCommandTest

hoholyin committed:
Update PPP and commandresult

hoholyin committed:
Add tests for bug fix

hoholyin committed:
Update UG

hoholyin committed:
Update developer's guide

hoholyin committed:
Fix language

hoholyin committed:
Add justification

shiyao821 committed:
fix: update SampleDataUtil

shiyao821 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main

hoholyin committed:
Fix bug (Dont allow exit during serve mode)

seeziyang committed:
Update PPP formatting

seeziyang committed:
Update PPP

Cronyxx committed:
update DeleteByIndexCommandTest

shiyao821 committed:
fix: asciidoctor css fix for Github pages

shiyao821 committed:
Merge branch 'UGDG' into gh-page-styling

shiyao821 committed:
Merge branch 'UGDG'

shiyao821 committed:
fix: PPP renamed

shiyao821 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main

shiyao821 committed:
fix: PPP and DG minor changes

shiyao821 committed:
fix: css files for Github Pages

shiyao821 committed:
fix: minor adjustment

shiyao821 committed:
fix: update UG DG

shiyao821 committed:
fix: update UG DG

shiyao821 committed:
fix: UI fixes

shiyao821 committed:
fix: UG command summary

shiyao821 committed:
fix: UG command summary

Cronyxx committed:
update DeleteBySerialNumberTest

shiyao821 committed:
Merge pull request #269 from hoholyin/portfolio

Changes to UG/DG

shiyao821 committed:
Merge pull request #268 from hoholyin/master

Fix bug (Unregister serving borrower)

hoholyin committed:
Merge pull request #271 from seeziyang/update-ppp

Update PPP and change all JavaFX versions to 11.0.0

hoholyin committed:
Merge pull request #270 from shiyao821/master

Update SampleDataUtil

linyutinglyt committed:
merge master UG

linyutinglyt committed:
update PPP

linyutinglyt committed:
update use case

linyutinglyt committed:
add use case to PPP

linyutinglyt committed:
change PPP name to lower case

linyutinglyt committed:
change PPP to github username

Cronyxx committed:
Merge pull request #191 from linyutinglyt/PPP

Yuting PPP

Cronyxx committed:
edit PPP

Cronyxx committed:
rename holyin's ppp

Cronyxx committed:
Merge pull request #276 from Cronyxx/add-ppp

Add ppp

shiyao821 committed:
Merge pull request #277 from shiyao821/master

Update PPP and Github Pages CSS

hoholyin committed:
Add link to Reposense

hoholyin committed:
Merge pull request #278 from hoholyin/portfolio

Add link to Reposense

hoholyin committed:
Edit shiyao PPP

hoholyin committed:
Merge pull request #279 from hoholyin/portfolio

Edit shiyao PPP

shiyao821 committed:
fix: UG serve command in wrong block

seeziyang committed:
Change set limits

hoholyin committed:
Fix Bug (Do not allow renewing books immediately after loaning

seeziyang committed:
Align photos to the left

seeziyang committed:
Move logging and configuration to below all our implementations

seeziyang committed:
Trying to fix photo alignments

hoholyin committed:
Update PPP

hoholyin committed:
Text fixes

hoholyin committed:
Merge pull request #281 from shiyao821/master

fix: UG serve command in wrong block

hoholyin committed:
Merge pull request #283 from seeziyang/fix-set-max-values

Change set limits

seeziyang committed:
Change back PPP formatting

seeziyang committed:
Update ppp woring

shiyao821 committed:
fix: github AboutUs image centralised

shiyao821 committed:
fix: SampleDataUtil Loans

shiyao821 committed:
Merge pull request #287 from shiyao821/gh-pages-styles

fix: github AboutUs image centralised

seeziyang committed:
Update UG

shiyao821 committed:
Merge branch 'gh-page-styling'

seeziyang committed:
Fix merge conflicts

shiyao821 committed:
fix: aboutus stylesheet test 2

shiyao821 committed:
Merge pull request #288 from shiyao821/gh-pages-styles

AboutUs centralise images test 2

hoholyin committed:
Styling

shiyao821 committed:
fix: aboutus stylesheet test 3

shiyao821 committed:
Merge pull request #289 from shiyao821/gh-pages-styles

fix: aboutus stylesheet test 3

seeziyang committed:
Fix image width

seeziyang committed:
Fix merge conflict

shiyao821 committed:
Merge pull request #285 from seeziyang/update-ppp

Update ppp

shiyao821 committed:
fix: aboutus stylesheet test 4

shiyao821 committed:
Merge pull request #290 from shiyao821/gh-pages-styles

fix: aboutus stylesheet test 4

shiyao821 committed:
fix: aboutus stylesheet test 5

seeziyang committed:
Update pay command in UG

shiyao821 committed:
Merge pull request #291 from shiyao821/gh-pages-styles

fix: aboutus stylesheet test 5

hoholyin committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main into portfolio

seeziyang committed:
Grammar changes

shiyao821 committed:
fix: MainApp startup sequence

hoholyin committed:
Bug fix to display return date

shiyao821 committed:
fix: Version Logger Info

shiyao821 committed:
fix: update code font color

shiyao821 committed:
fix: Undo bad change

shiyao821 committed:
Merge pull request #293 from shiyao821/gh-pages-styles

fix: update code font color 2

shiyao821 committed:
Merge pull request #286 from hoholyin/portfolio

Update documents and bug fixes

shiyao821 committed:
fix: image centralization

shiyao821 committed:
Merge pull request #294 from shiyao821/gh-pages-styles

fix: image centralization

shiyao821 committed:
fix: checkstyle

shiyao821 committed:
Merge remote-tracking branch 'origin/master'

shiyao821 committed:
fix: syntax

shiyao821 committed:
Merge pull request #295 from shiyao821/gh-pages-styles

fix: syntax

hoholyin committed:
Merge pull request #292 from shiyao821/master

Update SampleDataUtil, startup sequence, and Docs

seeziyang committed:
update dg

seeziyang committed:
Fix merge conflict

seeziyang committed:
update ug alignment

shiyao821 committed:
fix: syntax

shiyao821 committed:
Merge pull request #297 from shiyao821/gh-pages-styles

fix: syntax

seeziyang committed:
Refactor renew and return tests

seeziyang committed:
Merge branch 'master' into refactor-code

seeziyang committed:
Refactor renew

seeziyang committed:
Update renew ug

seeziyang committed:
Update renew msg

seeziyang committed:
Refactor code

seeziyang committed:
Fix merge conflict

seeziyang committed:
Update formatting

seeziyang committed:
Update dg

Cronyxx committed:
Merge pull request #298 from seeziyang/refactor-code

Refactor code for return and renew

seeziyang committed:
Update title

seeziyang committed:
Merge branch 'master' into update-ppp

Cronyxx committed:
update ug

seeziyang committed:
Add renew sequence diagram

seeziyang committed:
Update diagrams

Cronyxx committed:
implement ViewSettingsCommand

Cronyxx committed:
update ug

Cronyxx committed:
implement undoable toggleui

seeziyang committed:
Update DG my parts

Cronyxx committed:
update pay command

-Added reset command history to pay command.

seeziyang committed:
Update image size

seeziyang committed:
Update gh-pages css to not show  url links when printing

seeziyang committed:
Update dg

seeziyang committed:
Move fine to be above renew

seeziyang committed:
Fix grammar errors

seeziyang committed:
Update UiClassDiagram

hoholyin committed:
PPP

hoholyin committed:
Update styling

hoholyin committed:
Update PPP

seeziyang committed:
Add instructons for manual testing for loan, return, renew, and pay

seeziyang committed:
Updated use cases

hoholyin committed:
Style fixes

hoholyin committed:
Merge pull request #296 from seeziyang/update-ppp

Update ppp

hoholyin committed:
Update legend

Cronyxx committed:
update ppp

Cronyxx committed:
run checkstyle

hoholyin committed:
Merge branch 'master' into portfolio

Cronyxx committed:
Merge pull request #299 from hoholyin/portfolio

PPP

Cronyxx committed:
update user guide

Cronyxx committed:
update developer guide

hoholyin committed:
Merge pull request #300 from Cronyxx/update-documentation

Update documentation

hoholyin committed:
Update PPP

hoholyin committed:
Merge branch 'portfolio' of https://github.com/hoholyin/main into portfolio

hoholyin committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main into portfolio

seeziyang committed:
Fix typos

hoholyin committed:
summarized contributions

hoholyin committed:
First section

hoholyin committed:
Resize

hoholyin committed:
Restrict to 10page

hoholyin committed:
edit

hoholyin committed:
Edit

hoholyin committed:
Edit

hoholyin committed:
Edit

hoholyin committed:
edit

hoholyin committed:
page break

hoholyin committed:
page break

hoholyin committed:
edit

hoholyin committed:
edit

shiyao821 committed:
fix: Version number : 1.4.0

shiyao821 committed:
doc: DG Test Cases

hoholyin committed:
edit

shiyao821 committed:
doc: DG Use Cases

hoholyin committed:
Final fix

hoholyin committed:
Fix style

hoholyin committed:
add space

seeziyang committed:
Update ppp

seeziyang committed:
Merge pull request #301 from seeziyang/update-ppp

Fix typos

seeziyang committed:
Merge pull request #302 from hoholyin/portfolio

Summarize PPP

hoholyin committed:
Update Manual testing instructions

shiyao821 committed:
doc: DG features and ui.png

shiyao821 committed:
doc: PPP update

shiyao821 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/images/Ui.png

Cronyxx committed:
update dg

shiyao821 committed:
Fix: DG conflicts

shiyao821 committed:
Fix: DG EOF newline

shiyao821 committed:
Merge pull request #306 from shiyao821/master

DG Update

hoholyin committed:
Merge pull request #275 from Cronyxx/add-test-undoredo

Add tests for undo redo commands

hoholyin committed:
Merge branch 'master' into portfolio

Cronyxx committed:
Merge pull request #303 from hoholyin/portfolio

Update Manual testing instructions

linyutinglyt committed:
adding a borrower to be unregistered

linyutinglyt committed:
changing Borrower ID of borrower to be unregistered

linyutinglyt committed:
using Serve Mode exceptions

linyutinglyt committed:
Manual testing for serve and unregister

linyutinglyt committed:
PPP update

linyutinglyt committed:
resolve merge conflict

shiyao821 committed:
Update PPP - shiyao821

shiyao821 committed:
Merge pull request #308 from shiyao821/master

Update PPP - shiyao821

shiyao821 committed:
fix: File name change

Cronyxx committed:
add message for unused arguments

linyutinglyt committed:
merge from team master

linyutinglyt committed:
fix travis

shiyao821 committed:
fix: PPP EOF newline

shiyao821 committed:
Merge pull request #310 from shiyao821/master

Fix: syntax

shiyao821 committed:
Merge pull request #307 from linyutinglyt/PPP

Update DG for manual testing and PPP

shiyao821 committed:
fix: PPP - shiyao821

hoholyin committed:
Merge pull request #305 from Cronyxx/update-documentation

update dg

shiyao821 committed:
Merge pull request #309 from Cronyxx/bugfix-undoredo

add message for unused arguments

shiyao821 committed:
Merge pull request #311 from shiyao821/master

fix: PPP - shiyao821

hoholyin committed:
Completed javadocs

hoholyin committed:
Completed javadocs

shiyao821 committed:
fix: startup sequence displays load status

shiyao821 committed:
Merge pull request #312 from hoholyin/portfolio

Update javadocs

shiyao821 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F13-1/main

shiyao821 committed:
fix: DG language

shiyao821 committed:
fix: checkstyleMain

shiyao821 committed:
doc: PPP styling - shiyao821

shiyao821 committed:
doc: DG EOF newline

shiyao821 committed:
Merge pull request #313 from shiyao821/master

Add start sequence message + PPP updates

shiyao821 committed:
fix: data not found bug

shiyao821 committed:
Merge pull request #314 from shiyao821/master

fix: data not found bug

Cronyxx committed:
update documentation

Cronyxx committed:
Merge pull request #315 from Cronyxx/update-ppp

update documentation

Cronyxx committed:
make minor changes to diagram size

Cronyxx committed:
make minor changes to diagram size

Cronyxx committed:
Merge pull request #316 from Cronyxx/update-ppp

make minor changes to diagram size

Cronyxx committed:
change size of image in ppp

Cronyxx committed:
Merge pull request #317 from Cronyxx/update-ppp

Update ppp

Cronyxx committed:
minor adjustments to image size

Cronyxx committed:
Merge pull request #318 from Cronyxx/update-ppp

minor adjustments to image size

Cronyxx committed:
minor changes to image size

Cronyxx committed:
Merge pull request #319 from Cronyxx/update-ppp

minor changes to image size

seeziyang committed:
Remove corrupting data part in manual testing instructions

Cronyxx committed:
minor ppp update

Cronyxx committed:
Merge pull request #321 from Cronyxx/minor-ppp-update

minor ppp update

Cronyxx committed:
reduce word count

Cronyxx committed:
Merge pull request #322 from Cronyxx/minor-ppp-update

reduce word count

Cronyxx committed:
Merge pull request #320 from seeziyang/update-ppp

Remove corrupting data part in manual testing instructions

linyutinglyt committed:
adding alternatives to borrower feature

seeziyang committed:
Merge pull request #323 from linyutinglyt/PPP

adding alternatives to borrower feature on DG

Cronyxx committed:
update tests

shiyao821 committed:
Merge pull request #326 from Cronyxx/add-test-undoredo

update tests

shiyao821 committed:
doc: reduce PPP size

shiyao821 committed:
Merge pull request #327 from shiyao821/master

doc: reduce PPP size
